app/mutations/garden_crops/create_garden_crop.rb

```diff
@@ -1,51 +1,58 @@
 module GardenCrops
   class CreateGardenCrop < Mutations::Command
-    attr_writer :garden_crop
+    # attr_writer :garden_crop
 
     required do
       model :user
       string :garden_id
-    end
-
-    optional do
-      string :quantity
-      string :guide_id
-      string :crop_id
-      string :stage
-      date :sowed
-    end
-
-    def garden_crop
-      @garden_crop ||= GardenCrop.new
+      hash :attributes do
+        optional do
+          string :quantity
+          string :guide
+          string :crop
+          string :stage
+          date :sowed
+        end
+      end
     end
 
     def validate
       validate_guide
       validate_crop
+      validate_guide_or_crop
       validate_garden
       validate_permissions
     end
 
     def execute
-      set_params
-      garden_crop
+      @garden_crop ||= GardenCrop.new(attributes)
+      @garden_crop.garden = @garden
+      @garden_crop.save
+      @garden_crop
+    end
+
+    def validate_guide_or_crop
+      unless attributes[:guide] || attributes[:crop]
+        msg = "You need either a guide or a crop for the garden crop."
```

Comment:
Prefer single-quoted strings when you don't need string interpolation or special symbols.